Swamp Sneak/Assets/Scripts/Classes/Abstract Classes/Person.cs

```diff
@@ -1,8 +1,21 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using UnityEngine.AI;
 using UnityEngine;
 
 public abstract class Person : Entity {
 
+    //Variables shared across any type of person (Enemy & NPC)
```

Comment:
Player shouldn't have a NavMeshAgent, specifically it shouldn't be controlled by a navmeshagent. So this could break the Player or cause implementation issues.

I suggest modifying the hierarchy so that Person remains the parent level, but then NPC and Player are the ones that are direct descendants, then Enemy and Pedestrian **(New Class)** are descendants of NPC.

Then NPC can have this NavMeshAgent property and initialization.